nf_core/components/components_install.py

```diff
@@ -0,0 +1,127 @@
+import glob
+import json
+import logging
+import os
+import re
+
+import jinja2
+import questionary
+import rich
+
+import nf_core.modules.modules_utils
+import nf_core.utils
+from nf_core.modules.modules_repo import NF_CORE_MODULES_NAME
+
+log = logging.getLogger(__name__)
+
+
+def verify_sha(modules_repo, prompt, sha):
```

Comment:
I think this should be moved to a method on `ModulesRepo`. It doesn't have any specific component utility and always takes an instance of one as the first arg anyway.